src/backend/distributed/commands/index.c

```diff
@@ -208,6 +227,57 @@ PreprocessIndexStmt(Node *node, const char *createIndexCommand)
 }
 
 
+/*
+ * ErrorIfCannotGenerateDefaultIndexName errors out if we cannot generate a
+ * default index name for given CREATE INDEX command.
+ */
+static void
+ErrorIfCannotGenerateDefaultIndexName(IndexStmt *createIndexStatement)
+{
+	if (CanGenerateDefaultIndexName(createIndexStatement))
+	{
+		return;
+	}
+
+	ereport(ERROR, (errmsg("Cannot assign a default index name for CREATE INDEX command"),
```

Comment:
I'm a bit confused by this message and the naming of these functions. Postgres does not allow more than 32 (INDEX_MAX_KEYS) columns at all.

If you run a CREATE INDEX with more columns you get:
ERROR: cannot use more than 32 columns in an index

Should we just use the same message here?

Reply:
makes sense, I will update error message

Reply:
and the function names